src/components/PhotoFrame.tsx

```diff
@@ -241,6 +242,28 @@ const PhotoFrame = ({
         }
     }, [open]);
 
+    useEffect(() => {
+        const livePhotoVideo: HTMLVideoElement =
+            document.querySelector('video');
+        const livePhotoBtn: HTMLButtonElement =
+            document.querySelector('.live-photo-btn');
+        if (livePhotoVideo && livePhotoBtn) {
+            livePhotoBtn.addEventListener('mouseout', function () {
+                livePhotoVideo.pause();
+            });
+            livePhotoBtn.addEventListener('mouseover', function () {
+                livePhotoVideo.play().catch(() => {
+                    livePhotoVideo.pause();
+                });
+            });
+            livePhotoBtn.addEventListener('click', function () {
+                livePhotoVideo.play().catch(() => {
+                    livePhotoVideo.pause();
+                });
+            });
+        }
```

Comment:
we should also  remove the event listeners when the component dismounts using cleanup function
https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1

Reply:
Sure, will add it. I had thought that removal of the HTML element itself should also remove the event listeners associated with it.